Independent build

marvinmarnold opened the pull request:
**Why**
The hm-pktfwd build process is confusing. ~This is one of two proposals on how to structure the Dockerfile in order to improve things.~ We have decided to split up the build process by repo instead of having a mono-Dockerfile in hm-pktfwd.

**How**
Builds libloragw only and outputs to `/opt/output` to be retrieved by packet_forwarder and hm-pktfwd.

**References**
- Related to: https://github.com/NebraLtd/packet_forwarder/pull/2
- Related to: https://github.com/NebraLtd/sx1302_hal/pull/1
- Related to: https://github.com/NebraLtd/hm-pktfwd/pull/59
- Closed all-in-one proposal: https://github.com/NebraLtd/hm-pktfwd/pull/58
- Incorrectly opened here originally: https://github.com/Lora-net/lora_gateway/pull/172

marvinmarnold committed:
Revert "Change SPI port"

This reverts commit 13137065dc108a632a2dbb78c0533eea9425e78f.

marvinmarnold committed:
Revert "Remove normal file"

This reverts commit a09d936e88ac19b59f90bf62ff78f0b4ec1ef0c6.

marvinmarnold committed:
Revert "Re-add normal file back in."

This reverts commit 971c52e3e0f953102c0b057c9fff9b1df8a84d66.

marvinmarnold committed:
Revert "Change default SPI port back to 0.0"

This reverts commit 3d9f22f4b7649026d68443a3941dbe91036c1290.

marvinmarnold committed:
Revert "Add template file"

This reverts commit 7e635dbfc0331ee3c43f016628eda7bfe87e43d2.

marvinmarnold committed:
feat: build libloragw

marvinmarnold commented:
@shawaj thanks for pointing out the original wrong PR location. Both this and https://github.com/NebraLtd/packet_forwarder/pull/2 are meant to be drafts.

vpetersson commented:
LGTM so far structure wise.

vpetersson commented:
We should probably use releases here and reference the releases in `--from=lora_gateway` line in packet_forwarder? Wdyt?

marvinmarnold committed:
fix(build): set SPI_DEV_DIR and SPI_SPEED macros

Also improved logging.

marvinmarnold commented:
> We should probably use releases here and reference the releases in `--from=lora_gateway` line in packet_forwarder? Wdyt?

Using releases makes sense but I'm not sure how to do that automagically. For the moment, I'm just [using commit hashes](https://github.com/NebraLtd/packet_forwarder/pull/2/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R24) like we do in helium-miner-software.

shawaj commented:
> > We should probably use releases here and reference the releases in `--from=lora_gateway` line in packet_forwarder? Wdyt?
> 
> Using releases makes sense but I'm not sure how to do that automagically. For the moment, I'm just [using commit hashes](https://github.com/NebraLtd/packet_forwarder/pull/2/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R24) like we do in helium-miner-software.

We can automatically tag and release easily as per 

https://github.com/NebraLtd/hm-miner/blob/79923ea6ab73a69c446a99cbad658e6453d97c9d/.github/workflows/check-latest-release.yml#L44-L59

But I guess this would also require a semver action such as:
https://github.com/marketplace/actions/git-semantic-version

Otherwise we would have to manually tag/release.

Maybe easier to just use the commit SHA like you have already?

shawaj commented on compile_libloragw.sh:
```diff
@@ -0,0 +1,40 @@
+    compile_libloragw_for_spi_bus spidev32766.0
+}
+
+copy_reset_script() {
```
Probably don't need this 

Can just use the same reset script as for the sx1302 instead of copying multiple

See also https://github.com/NebraLtd/hm-pktfwd/pull/59#discussion_r733259913

marvinmarnold replied:
I do see some small differences between the scripts but I'm unclear how significant they are. I agree we should only copy one file if the differences are not important.

1. [sx1301 sets input during init](https://github.com/NebraLtd/lora_gateway/blob/971c52e3e0f953102c0b057c9fff9b1df8a84d66/reset_lgw.sh#L37) but [sx1302 doesn't](https://github.com/NebraLtd/sx1302_hal/pull/1/files#diff-bd5477175815818c1a48d879d7debc4d6c882e762a8715872cf2f193371a6065L36)

2. [sx1301 stop doesn't reset](https://github.com/NebraLtd/lora_gateway/blob/971c52e3e0f953102c0b057c9fff9b1df8a84d66/reset_lgw.sh#L54) but [sx1302 does](https://github.com/NebraLtd/sx1302_hal/pull/1/files#diff-bd5477175815818c1a48d879d7debc4d6c882e762a8715872cf2f193371a6065R54)

Want to make the call @shawaj?

shawaj replied:
1. the reason they are doing that is because they seem to be using GPIO7 for both the reset pin and as the SPI chip select pin. I don't understand why they did this - it is a bizarre choice. But we do not need to set any inputs on the reset pin as it is only used for reset and nothing else

2. iot_sk_init() in sx1301 folder is the same function essentially as init() and reset() together in the sx1302 function. so this is doing the same thing just in separate functions.

So I would say the sx1302 one makes more sense to keep

marvinmarnold replied:
1. Cool
2. @shawaj Not sure we are talking about the same thing. The relevant code is in a case statement to handle `start` and `stop` commands differently. I agree that `iot_sk_init()` vs `init()`+`reset()` are functionally the same for `start`. But for `stop`, the sx1301 version is missing the call to reset/init all together. That said, I'm OK using the sx1302 code only and will make that change.

shawaj replied:
@marvinmarnold i don't know exactly why there is a need to reset before the term() command - to be honest I think that is unecesary in sx1302 - but it also will not harm so may as well leave it in (for either sx1301 or sx1302)

marvinmarnold commented:
> We can automatically tag and release easily as per
> 
> https://github.com/NebraLtd/hm-miner/blob/79923ea6ab73a69c446a99cbad658e6453d97c9d/.github/workflows/check-latest-release.yml#L44-L59
> 
> But I guess this would also require a semver action such as: https://github.com/marketplace/actions/git-semantic-version
> 
> Otherwise we would have to manually tag/release.
> 
> Maybe easier to just use the commit SHA like you have already?

Let's prioritize getting the current work over the line. The steps you laid out should work. Created a [new issue](https://github.com/NebraLtd/hm-pktfwd/issues/61) to track this work.

marvinmarnold committed:
fix: remove reset_lgw from output

This script duplicates logic in the sx1302_hal version.
That version will be used instead.
